helix-view/src/document.rs

```diff
@@ -431,15 +431,16 @@ impl Document {
     // TODO: async fn?
     /// Create a new document from `path`. Encoding is auto-detected, but it can be manually
     /// overwritten with the `encoding` parameter.
-    pub fn open(
-        path: PathBuf,
+    pub fn open<P: AsRef<Path>>(
```

Comment:
We need to be careful about using generics on functions longer than a few lines. I think we should either use &Path here, or have open be a wrapper that takes AsRef and then open_impl takes &Path. 

See the "Inlining in Practice" section here:
https://matklad.github.io/2021/07/09/inline-in-rust.html